seL4-platforms/builds.py

```diff
@@ -820,16 +820,16 @@ def match_dict(build: Build, f):
                 if build.get_mode() not in v:
                     return False
             elif k == 'mcs':
-                if v != '' and not build.is_mcs():
+                if (v == '') == build.is_mcs():
                     return False
             elif k == 'smp':
-                if v != '' and not build.is_smp():
+                if (v == '') == build.is_smp():
                     return False
             elif k == 'hyp':
-                if v != '' and not build.is_hyp():
+                if (v == '') == build.is_hyp():
```

Comment:
This isn't really my area, but the previous test said "if there is a value here but it this isn't hyp", and now it says "if there is a value, it must be hyp, otherwise it can't be hyp". That would imply that for all of these, any non-empty value is always an enabling value? I.e. asking for 1 domain implies that domains are enabled?

Reply:
Yes, the distinction here is just empty string or any non-empty string.

The test config does not support setting how many domains, it just sets the env variable `DOMAINS` or not. The individual tests then have to figure out what they want to do with that.